functions/ocr/app/test/index.test.js

```diff
@@ -1,438 +1,271 @@
-      language: 'ja'
-    });
-    ocrSample.sample.processImage(context, {
-      bucket: bucket,
-      name: name
-    });
-  });
-
-  it('translateText fails without text', function () {
-    var expectedMsg = 'Text not provided. Make sure you have a ' +
-      '"text" property in your request';
-    var context = getMockContext();
-
-    getSample().sample.translateText(context, {});
-
-    assert.equal(context.failure.calledOnce, true);
-    assert.equal(context.failure.firstCall.args[0], expectedMsg);
-    assert.equal(context.success.called, false);
-    assert.equal(console.error.called, true);
-  });
-
-  it('translateText fails without a filename', function () {
-    var expectedMsg = 'Filename not provided. Make sure you have a ' +
-      '"filename" property in your request';
-    var context = getMockContext();
-
-    getSample().sample.translateText(context, {
-      text: text
-    });
-
-    assert.equal(context.failure.calledOnce, true);
-    assert.equal(context.failure.firstCall.args[0], expectedMsg);
-    assert.equal(context.success.called, false);
-    assert.equal(console.error.called, true);
+    const sample = getSample();
+
+    return sample.program.processImage(event)
+      .then(() => {
+        assert.equal(console.log.callCount, 4);
+        assert.deepEqual(console.log.getCall(0).args, [`Looking for text in image ${fileName}`]);
+        assert.deepEqual(console.log.getCall(1).args, [`Extracted text from image (${text.length} chars)`]);
+        assert.deepEqual(console.log.getCall(2).args, [`Detected language "ja" for ${fileName}`]);
+        assert.deepEqual(console.log.getCall(3).args, [`File ${event.data.name} processed.`]);
+      });
   });
 
-  it('translateText fails without a lang', function () {
-    var expectedMsg = 'Language not provided. Make sure you have a ' +
-      '"lang" property in your request';
-    var context = getMockContext();
-
-    getSample().sample.translateText(context, {
-      text: text,
-      filename: filename
-    });
-
-    assert.equal(context.failure.calledOnce, true);
-    assert.equal(context.failure.firstCall.args[0], expectedMsg);
-    assert.equal(context.success.called, false);
-    assert.equal(console.error.called, true);
-  });
-
-  it('translateText handles translation error', function (done) {
-    var expectedMsg = 'error';
-    var context = {
-      success: assert.fail,
-      failure: function () {
-        assert.equal(context.failure.calledOnce, true);
-        assert.equal(context.failure.firstCall.args[0], expectedMsg);
-        assert.equal(context.success.called, false);
-        assert.equal(console.error.calledWith(expectedMsg), true);
-        done();
+  it(`translateText fails without text`, () => {
+    const error = new Error(`Text not provided. Make sure you have a "text" property in your request`);
+    const event = {
+      data: {
+        data: Buffer.from(JSON.stringify({})).toString(`base64`)
```

Comment:
I want it to be clear that you can use an actual object when constructing the message payload, and that the object must be stringifed.